### PR TITLE
Add redundans

### PR DESCRIPTION
**Pull request by Dfupa**
Add redundans recipe.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by Dfupa** [comment]
@BiocondaBot please add label

**Comment by Dfupa** [comment]
@BiocondaBot please add label

**Comment by Dfupa** [comment]
Hi @bioconda/core let me know whether this request could be reviewed and successfully merged. Thanks!

**Comment by mencian** [comment]
Hello @Dfupa, thanks for your contribution and I apologize for just now getting to this.
Is it possible to package a binary release with all the git submodules included? Bioconda does not recommend downloading/cloning source files from versioning systems (see https://bioconda.github.io/contributor/linting.html#uses-vcs-url) as
 1) commits made after the time of packaging may cause the entire build to be unstable and 
 2) would greatly simplify the build recipe.

File: recipes/redundans/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  home: https://github.com/Gabaldonlab/redundans/
+  license: GNU General Public License v3.0
+  license_file: LICENSE
+  summary: 'Redundans is a pipeline that assists an assembly of heterozygous/polymorphic genomes.'
+
+extra:
+  identifiers:
+    - biotools:redundans
+  recipe-maintainers:
+    - Diego Fuentes Palacios
```
Comment (mencian) [inline recipes/redundans/meta.yaml:72]:
```suggestion
    - Dfupa
```

File: recipes/redundans/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - r-scales >=1.1.1
+    
+test:
+  commands:
+    - redundans.py --version
+
+
+about:
+  home: https://github.com/Gabaldonlab/redundans/
+  license: GNU General Public License v3.0
```
Comment (mencian) [inline recipes/redundans/meta.yaml:64]:
```suggestion
  license: GPL-3.0-or-later
  license_family: GPL3
```

File: recipes/redundans/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+requirements:
+  build:
+    - make
+    - zlib
+    - tar
+    - git
+    - wget
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
```
Comment (mencian) [inline recipes/redundans/meta.yaml:30]:
```suggestion
    - {{ compiler('cxx') }}
    - {{ compiler('c') }}
```

File: recipes/redundans/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  host:
+    - python >=3.8,<3.11
+    - pip
+    - zlib
+    - pillow
+    - git
+    - perl
+    - make
```
Comment (mencian) [inline recipes/redundans/meta.yaml:39]:
```suggestion
    - zlib
    - pillow
    - perl
```

File: recipes/redundans/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
+{% set name = "Redundans" %}
+{% set version = "2.0.1" %}
+{% set sha256hash = "85a1937015aa0af44a5402761356c4098b085ee7233fb1b7cf8254e2ca0f0e54" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Gabaldonlab/redundans/archive/v2.01.tar.gz
```
Comment (mencian) [inline recipes/redundans/meta.yaml:10]:
```suggestion
  url: https://github.com/Gabaldonlab/redundans/archive/refs/tags/v{{ version }}.tar.gz
```

**Review by mencian — COMMENTED** [review]
Thanks for the doing the heavy-lifting! The recipe looks much better now. If you do not mind, I will take it from here and clean up the recipe.

File: recipes/redundans/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Gabaldonlab/redundans/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256hash }}
+
+build:
+  number: 0
+  skip: True  # [osx]
```
Comment (mencian) [inline recipes/redundans/meta.yaml:15]:
```suggestion
  number: 0
```

File: recipes/redundans/build.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+#Define the install folder and binary folder and create them
+INSTALL_FOLDER=$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM
+BIN_FOLDER=$PREFIX/bin
+mkdir -p $INSTALL_FOLDER && mkdir -p $BIN_FOLDER
+
+#Copying files to proper installation folder
+chmod u+x redundans.py #To avoid permission issues
+
+#Avoid pathing issues with SSPACE pl script
+sed -i 's|bin/SSPACE/SSPACE_Standard_v3.0.pl|SSPACE/SSPACE_Standard_v3.0.pl|g' redundans.py
```
Comment (mencian) [inline recipes/redundans/build.sh:12]:
```suggestion
sed -i.bak 's|bin/SSPACE/SSPACE_Standard_v3.0.pl|SSPACE/SSPACE_Standard_v3.0.pl|g' redundans.py
```

**Review by mencian — COMMENTED** [review]
A couple of things I have missed:
1. The version string is `2.01`, not `2.0.1`, so I have suggested this change for the `version` variable.
2. If you would like the recipe to build on OSX, `sed -i` will need to be changed to `sed -i.bak` for compatibility purposes.

File: recipes/redundans/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Gabaldonlab/redundans/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256hash }}
+
+build:
+  number: 0
```
Comment (mencian) [inline recipes/redundans/meta.yaml:14]:
```suggestion
  number: 0
  skip: True  # [osx]
```

**Review by mencian — COMMENTED** [review]
Last change (hopefully) to revert back to skipping OSX build due to snap-aligner not being available on OSX...

**Review by mencian — APPROVED** [review]

**Comment by Dfupa** [comment]
@mencian 

For the next release I can try to provide the binary of snap-aligner and try again for a OSX version if that's all it takes

**Comment by mencian** [comment]
Sounds good - the package _should_ build on OSX without issue once the snap-aligner dependency is resolved. If you need assistance with publishing future releases on Bioconda, please do not hesistate to ping me!

**Pull request merged**